PublicMapping/district-builder-2
# Allow for selecting chamber/number of district when creating project

## Overview

Allow for selecting chamber/number of district when creating project

### Demo
![district_builder_chamber2](https://user-images.githubusercontent.com/2926237/80633174-58b70080-8a26-11ea-8619-6acc81da4f59.gif)

### Notes
Chamber creation is manual (no way to do it via the API or UI).

This architecture should allow us to create any kind of chambers we want with any name or number of districts in the future.

## Testing Instructions
* ~Delete any projects you may have created previously since a chamber is now required (`delete from project;`)~
* Run migrations
* Create some regions and chambers to choose from when creating projects. Here's a database dump of what I have used for testing:
```
INSERT INTO public.region_config (id, name, country_code, region_code, s3_uri, version) VALUES ('4534e423-bacd-4634-b581-bd289fc3d4b1', 'Pennsylvania', 'US', 'PA', 's3://test/ing/123456', '2020-04-20 17:11:54.156542+00');
INSERT INTO public.region_config (id, name, country_code, region_code, s3_uri, version) VALUES ('59a7b1b5-5f4c-4164-997c-579eee2e77ed', 'Virginia', 'US', 'VA', 's3://test/ing/va12345', '2020-04-22 15:31:06.459685+00');
INSERT INTO public.region_config (id, name, country_code, region_code, s3_uri, version) VALUES ('76fbf118-e3ec-432e-9a23-0f5b1649b4bf', 'New York', 'US', 'NY', 's3://test/ing/ny12345', '2020-04-23 15:41:44.953+00');
INSERT INTO public.region_config (id, name, country_code, region_code, s3_uri, version) VALUES ('451c80cc-c990-4c25-aa4f-dbc0c60875e0', 'New York', 'US', 'NY', 's3://test/ing/ny123456', '2020-04-27 18:51:52.499744+00');
INSERT INTO public.region_config (id, name, country_code, region_code, s3_uri, version) VALUES ('07a71eaa-df6a-4f08-bce6-d8d6e4fb8585', 'Nebraska', 'US', 'NE', 's3://test/ing/ne123', '2020-04-29 13:19:53.322618+00');

INSERT INTO public.chamber (id, name, number_of_districts, region_config_id) VALUES ('3d6f967f-2aad-4e33-a11c-366e7a19bbba', 'House of Representatives', 18, '4534e423-bacd-4634-b581-bd289fc3d4b1');
INSERT INTO public.chamber (id, name, number_of_districts, region_config_id) VALUES ('28d8b26a-ffdf-437d-ad46-a4d6d6434459', 'State Senate', 50, '4534e423-bacd-4634-b581-bd289fc3d4b1');
INSERT INTO public.chamber (id, name, number_of_districts, region_config_id) VALUES ('754361e8-b289-46f3-8bc4-8f2227345b28', 'State House', 203, '4534e423-bacd-4634-b581-bd289fc3d4b1');
INSERT INTO public.chamber (id, name, number_of_districts, region_config_id) VALUES ('377ce924-b629-476b-b0a3-2e5b6a8fbfc6', 'House of Representatives', 3, '07a71eaa-df6a-4f08-bce6-d8d6e4fb8585');
INSERT INTO public.chamber (id, name, number_of_districts, region_config_id) VALUES ('2904d5af-2dfc-4a54-be0b-da58207e326e', 'State Legislature', 49, '07a71eaa-df6a-4f08-bce6-d8d6e4fb8585');
```
* Create a project as per the demo and ensure the correct region and number of districts is shown on the home page after successful creation
* Make sure you test both selecting a chamber and specifying a custom number of districts

Closes #84
